node/src/index.ts

```diff
@@ -89,13 +89,17 @@ export class Table {
 export class Query {
   private readonly _tbl: any
   private readonly _query_vector: number[]
-  private readonly _limit: number
+  private _limit: number
   private readonly _refine_factor?: number
   private readonly _nprobes: number
   private readonly _columns?: string[]
   private readonly _where?: string
   private readonly _metric = 'L2'
 
+  set limit(value: number) {
```

Comment:
should we keep the constructor to be the first method in the class?

And why only limit is not readonly? 

Also what's the reason that use `_` as prefix for parameters here?